### PR TITLE
node work...

### LINKED ISSUE #7: manifest-validator.js Depends on Document
I investigated what would be needed to use the validator in a gulp script. Reading a manifest file into a string and passing it to parse() is pretty straight forward. Unfortunately, manifest-parser.js has references to the document object. I investigated using node-dom to make the script think it was in a browser, but that entails having an otherwise useless html file in memory.

### PR DESCRIPTION
**Pull request by samccone**
:tada: Happy Friday, here is some node work to top your evening off :tophat: 
- setup global document shim in common js
- lock down mocha version
- fixup mocha usage 
- lock down node range to support basically everyone
- setup jscs and fixup file to conform

:dancer: 

---

Fixes #7

### CODE REVIEW

**Comment by samccone** [comment]
:runner: fixing up the CI failures

**Comment by samccone** [comment]
kk I am investigating how we can get node 0.12 and 0.10 to work... however because jsdom 7+ only works in node 4.x+ I am going to have to do some funky work with the install process... totally solvable however should not block this work from landing :airplane:

**Comment by mounirlamouri** [comment]
The problem is that Ubuntu users are stuck with 0.10.25 if they use the out of the box configuration. I think requiring node-4.x+ adds a large overhead. I would prefer if we could address this issue before merging this PR. Also, maybe you could upgrade the nodejs requirements in the same commit where you add jsdom? Otherwise, the link between these two isn't clear.

You might want to split non-jsdom stuff out of this PR so we can land them though.

**Comment by samccone** [comment]
(ps working on this :) )

**Comment by samccone** [comment]
woot passing :)

Had to make a new lib for this to be possible ha!
https://github.com/samccone/engine-deps

:tada:

**Comment by mounirlamouri** [comment]
I think you need to rebase. LGTM otherwise.

**Comment by samccone** [comment]
:+1: all set @mounirlamouri :wink: :wine_glass:

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
-- node
-- '0.10'
+  - "0.10"
+  - "0.12"
+  - "4"
+  - "5"
 branches:
   only:
-  - gh-pages
+    - gh-pages
```
Comment (mounirlamouri) [inline .travis.yml:9]: Hmm, I removed the "branches" list from .travis.yml.

Comment (samccone) [inline .travis.yml:9]: gah one moment

**Comment by samccone** [comment]
i need to close and reopen since I am changing the PR target... a fun limitation of github

**Pull request closed without merging**